### ISSUE TITLE
[Bug] Unhandled Environment exception. - System.ObjectDisposedException: Cannot access a disposed object.

### ISSUE THREAD

**Issue by vishnupadakath** [xamarin/Xamarin.Forms#9412]
Hi,

I have list view in my android app,   with a list of items. The list view has an item tapped event. Every time i tap the item , message box appears asking if you actually want to select the item . If clicked ok, then goes through further programs and again shows another message box showing item selected. 

Occasionally, i get the error : Unhandled Environment exception. - System.ObjectDisposedException: Cannot access a disposed object. This is usually after i select  "ok" for the first message box and  before the second message box appears .  my app crashes whenever this happens.   I Think this is similar to the issue in the link 

https://github.com/xamarin/Xamarin.Forms/issues/7325. 

logs from Android app 

2-03 17:27:00.964	Blackview A60Pro	Error	26847	XenplateDC	Object name: 'Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer'.
  at Java.Interop.JniPeerMembers.AssertSelf (Java.Interop.IJavaPeerable self) [0x00029] in <286213b9e14c442ba8d8d94cc9dbec8e>:0 
  at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeNonvirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x00000] in <286213b9e14c442ba8d8d94cc9dbec8e>:0 
  at Android.Widget.TextView.set_TextFormatted (Java.Lang.ICharSequence value) [0x00029] in <95d915721ad748108ff096dce8f8f32c>:0 
  at Android.Widget.TextView.set_Text (System.String value) [0x0000d] in <95d915721ad748108ff096dce8f8f32c>:0 
  at Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer.UpdateText () [0x00081] in D:\a\1\s\Xamarin.Forms.Platform.Android\FastRenderers\LabelRenderer.cs:368 
  at Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer.OnElementPropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs e) [0x000ee] in D:\a\1\s\Xamarin.Forms.Platform.Android\FastRenderers\LabelRenderer.cs:272 
  at (wrapper delegate-invoke) <Module>.invoke_void_object_PropertyChangedEventArgs(object,System.ComponentModel.PropertyChangedEventArgs)
  at Xamarin.Forms.BindableObject.OnPropertyChanged (System.String propertyName) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:211 
  at Xamarin.Forms.Element.OnPropertyChanged (System.String propertyName) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:359 
  at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) [0x00114] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:443 
  at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x00173] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:379 
  at Xamarin.Forms.BindingExpression.ApplyCore (System.Object sourceObject, Xamarin.Forms.BindableObject target, Xamarin.Forms.BindableProperty property, System.Boolean fromTarget) [0x00220] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:167 
  at Xamarin.Forms.BindingExpression.Apply (System.Object sourceObject, Xamarin.Forms.BindableObject target, Xamarin.Forms.BindableProperty property) [0x0006b] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:78 
  at Xamarin.Forms.Binding.Apply (System.Object context, Xamarin.Forms.BindableObject bindObj, Xamarin.Forms.BindableProperty targetProperty, System.Boolean fromBindingContextChanged) [0x00057] in D:\a\1\s\Xamarin.Forms.Core\Binding.cs:131 
  at Xamarin.Forms.BindableObject.ApplyBindings (System.Boolean skipBindingContext, System.Boolean fromBindingContextChanged) [0x00041] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:462 
  at Xamarin.Forms.BindableObject.SetInheritedBindingContext (Xamarin.Forms.BindableObject bindable, System.Object value) [0x0005a] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:191 
  at Xamarin.Forms.Element.SetChildInheritedBindingContext (Xamarin.Forms.Element child, System.Object context) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:511 
  at Xamarin.Forms.TemplatedView.SetChildInheritedBindingContext (Xamarin.Forms.Element child, System.Object context) [0x00008] in D:\a\1\s\Xamarin.Forms.Core\TemplatedView.cs:69 
  at Xamarin.Forms.Element.<OnBindingContextChanged>b__78_0 (Xamarin.Forms.BindableObject child, System.Object bc) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:314 
  at Xamarin.Forms.BindableObjectExtensions.PropagateBindingContext[T] (Xamarin.Forms.BindableObject self, System.Collections.Generic.IList`1[T] children, System.Action`2[T1,T2] setChildBindingContext)

Your help will be much appreciated 

Thanks
Vishnu

**Comment by samhouts** [comment]
@vishnupadakath Please use the entire template when submitting an issue. What version of Xamarin.Forms are you using? What version of android are you targeting, and what version are you deploying to?  Can you please attach a small project that demonstrates this issue? Thanks!

**Comment by vishnupadakath** [comment]
Hi,

Thanks for looking in to this . 
Xamarin form version : 4.2.0.848062
Android Target Framework : Android 9.0(API Level 28- Pie) 
Android Deployed -: same as target.  
.net framework Standard 2.0 

I prepared a sample project with just the pages  that causes the issue. However I am not able to reproduce this issue in the sample.  Do you still want the sample? 

Thanks
Vishnu

**Comment by samhouts** [comment]
@vishnupadakath Yes, it may help us trace where the issue is. Thanks!

**Comment by vishnupadakath** [comment]
Hi ,

Please find attached the sample project . I created this by copying some of the code from the live app. Please ignore the UWP and IOS project . Apparently the crash do not occur in iOS. the issue is only in android. 

Although, the sample project do not break , following are the steps that crashed the live app
1) Launch the app
2) on first screen select one item in the list .
3) Select one of the  device .
4) Click ok for the first pop up.
5) Click ok for the second pop up.
6) go back to the first page. Do steps 2 - 5 . 

If you add more than one devices by following the above steps , the app crashes after step 4 .

Please let me know if you need any further information. 

Thanks
Vishnu
[App2.zip](https://github.com/xamarin/Xamarin.Forms/files/4161055/App2.zip)

**Comment by samhouts** [comment]
@vishnupadakath Thank you for the sample code. I see that it is referencing 4.2.0.848062. Can you please try updating to the latest version to see if the problem persists? We have resolved many similar issues since 4.2.0. Thanks!

**Comment by vishnupadakath** [comment]
Hi,

Many thanks for your suggestion.  Using the latest version of Xamarin fixed the issue. Our testing is still in progress . Hopefully the app does not  crash ! 

Regards
Vishnu

**Comment by samhouts** [comment]
@vishnupadakath Wonderful! We'll wait to hear back from you. Thanks!

**Comment by furiozo** [comment]
Getting this on XF 4.4.0.991265 -  latest iOS. Tried the latest, still the same.

Stack trace:
`2020-02-08 01:27:04.306090+0200 IRA.iOS[24737:184880] Unhandled managed exception: Cannot access a disposed object.
Object name: 'PrimaryToolbarItem'. (System.ObjectDisposedException)
  at Foundation.NSObject.get_SuperHandle () [0x00012] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.10.0.17/src/Xamarin.iOS/Foundation/NSObject2.cs:449 
  at UIKit.UIBarButtonItem.set_Image (UIKit.UIImage value) [0x00033] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.10.0.17/src/Xamarin.iOS/UIBarButtonItem.g.cs:652 
  at Xamarin.Forms.Platform.iOS.ToolbarItemExtensions+PrimaryToolbarItem.UpdateIconAndStyle () [0x00031] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Extensions\ToolbarItemExtensions.cs:84 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1021 
  at Foundation.NSAsyncSynch
ronizationContextDispatcher.Apply () [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.10.0.17/src/Xamarin.iOS/Foundation/NSAction.cs:178 
--- End of stack trace from previous location where exception was thrown ---



  
  2020-02-08 01:44:44.737620+0200 IRA.iOS[27782:209679] Unhandled managed exception: Cannot access a disposed object.
Object name: 'ParentingViewController'. (System.ObjectDisposedException)
  at Foundation.NSObject.get_SuperHandle () [0x00012] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.10.0.17/src/Xamarin.iOS/Foundation/NSObject2.cs:449 
  at UIKit.UIViewController.get_NavigationItem () [0x0002a] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.10.0.17/src/Xamarin.iOS/UIViewController.g.cs:2303 
  at Xamarin.Forms.Platform.iOS.NavigationRenderer+<>c__DisplayClass76_0.<SetMasterLeftBarButton>b__0 (UIKit.UIImage icon) [0x0002c] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\NavigationRenderer.cs:761 
  at Xamarin.Forms.Platform.iOS.ImageElementManager.ApplyNativeImageAsync (Xamarin.Forms.BindableObject bindable, Xamarin.Forms.BindableProperty imageSourceProperty, System.Action`1[T] onSet, System.Action`1[T] onLoading, System.Threading.CancellationToken cancellationToken) [0x000fd] in D:\a
\1\s\Xamarin.Forms.Platform.iOS\Renderers\ImageElementManager.cs:281 
  at Xamarin.Forms.Platform.iOS.NavigationRenderer.SetMasterLeftBarButton (UIKit.UIViewController containerController, Xamarin.Forms.MasterDetailPage masterDetailPage) [0x0007f] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\NavigationRenderer.cs:747 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1021 
  at Foundation.NSAsyncSynchronizationContextDispatcher.Apply () [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.10.0.17/src/Xamarin.iOS/Foundation/NSAction.cs:178 
--- End of stack trace from previous location where exception was thrown ---`

**Comment by samhouts** [comment]
@furiozo Based on your stack trace, that is a different problem. Can you please open a new issue and attach a sample project, if possible? Thanks!

**Comment by furiozo** [comment]
Sure thanks.

**Comment by RonnyBansemer** [comment]
We're currently facing the same issue using XF 4.4.0.991640. It occours when the MainThread, where the text changes are invoked, is on high load. 
The issue is located at https://github.com/xamarin/Xamarin.Forms/blob/239ba5046acc37259f3135bc33dd1b27bf72c8f3/Xamarin.Forms.Platform.Android/FastRenderers/LabelRenderer.cs at line 406 and caused by a missing ZeroPointer-Check.

```
[MonoDroid] UNHANDLED EXCEPTION:
[MonoDroid] System.ObjectDisposedException: Cannot access a disposed object.
[MonoDroid] Object name: 'Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer'.
[MonoDroid]   at Java.Interop.JniPeerMembers.AssertSelf (Java.Interop.IJavaPeerable self) [0x00029] in <af1c6fdcff1a4da4a0e44a8cdef352da>:0
[MonoDroid]   at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeNonvirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x00000] in <af1c6fdcff1a4da4a0e44a8cdef352da>:0
[MonoDroid]   at Android.Widget.TextView.set_TextFormatted (Java.Lang.ICharSequence value) [0x00029] in <e98afaf88af5496ea979d99b06e991b9>:0
[MonoDroid]   at Android.Widget.TextView.set_Text (System.String value) [0x0000d] in <e98afaf88af5496ea979d99b06e991b9>:0
[MonoDroid]   at Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer.UpdateText () [0x000f3] in D:\a\1\s\Xamarin.Forms.Platform.Android\FastRenderers\LabelRenderer.cs:406
[MonoDroid]   at Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer.OnElementPropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs e) [0x00123] in D:\a\1\s\Xamarin.Forms.Platform.Android\FastRenderers\LabelRenderer.cs:287
[MonoDroid]   at (wrapper delegate-invoke) <Module>.invoke_void_object_PropertyChangedEventArgs(object,System.ComponentModel.PropertyChangedEventArgs)
[MonoDroid]   at Xamarin.Forms.BindableObject.OnPropertyChanged (System.String propertyName) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:229
[MonoDroid]   at Xamarin.Forms.Element.OnPropertyChanged (System.String propertyName) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:353
[MonoDroid]   at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) [0x00114] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:461
[MonoDroid]   at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x00173] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:397
[MonoDroid]   at Xamarin.Forms.BindingExpression.ApplyCore (System.Object sourceObject, Xamarin.Forms.BindableObject target, Xamarin.Forms.BindableProperty property, System.Boolean fromTarget) [0x00220] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:156
[MonoDroid]   at Xamarin.Forms.BindingExpression.Apply (System.Boolean fromTarget) [0x0003e] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:52
[MonoDroid]   at Xamarin.Forms.BindingExpression+BindingExpressionPart.<PropertyChanged>b__49_0 () [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:741
[MonoDroid]   at Xamarin.Forms.BindingExpression+BindingExpressionPart.PropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs args) [0x000be] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:752
[MonoDroid]   at Xamarin.Forms.BindingExpression+WeakPropertyChangedProxy.OnPropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs e) [0x00012] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:645
[MonoDroid]   at (wrapper delegate-invoke) <Module>.invoke_void_object_PropertyChangedEventArgs(object,System.ComponentModel.PropertyChangedEventArgs)
[MonoDroid]   at SomeApp.Models.BookingButtonFacade.set_ReceiptAmount (System.Int32 value) [0x0000d] in /Users/vagrant/Projects/KORONA.express/KORONA.express-definitions/Models/BookingButtonFacade.cs:17
[MonoDroid]   at SomeApp.ViewModels.Abstract.BookingButtonPageViewModelBase+<>c__DisplayClass21_0.<TriggerCollectionChanged>b__3 (SomeApp.Models.BookingButtonFacade x) [0x00000] in /Users/vagrant/Projects/KORONA.express/KORONA.express/ViewModels/Abstract/BookingButtonPageViewModelBase.cs:139
[MonoDroid]   at Xamarin.Forms.Internals.EnumerableExtensions.ForEach[T] (System.Collections.Generic.IEnumerable`1[T] enumeration, System.Action`1[T] action) [0x00010] in D:\a\1\s\Xamarin.Forms.Core\EnumerableExtensions.cs:58
[MonoDroid]   at SomeApp.ViewModels.Abstract.BookingButtonPageViewModelBase.TriggerCollectionChanged (System.Collections.ObjectModel.ObservableCollection`1[T] bookingButtons, System.Collections.Specialized.NotifyCollectionChangedAction action, System.Collections.IList newItems, System.Collections.IList oldItems) [0x00063] in /Users/vagrant/Projects/KORONA.express/KORONA.express/ViewModels/Abstract/BookingButtonPageViewModelBase.cs:138
[MonoDroid]   at SomeApp.ViewModels.Abstract.BookingButtonPageViewModelBase+<>c__DisplayClass20_0.<ProductsOnReceiptCollectionChanged>b__0 () [0x00001] in /Users/vagrant/Projects/KORONA.express/KORONA.express/ViewModels/Abstract/BookingButtonPageViewModelBase.cs:124
[MonoDroid]   at Xamarin.Essentials.MainThread.BeginInvokeOnMainThread (System.Action action) [0x00007] in D:\a\1\s\Xamarin.Essentials\MainThread\MainThread.shared.cs:16
[MonoDroid]   at SomeApp.ViewModels.Abstract.BookingButtonPageViewModelBase.ProductsOnReceiptCollectionChanged (System.Object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e) [0x00015] in /Users/vagrant/Projects/KORONA.express/KORONA.express/ViewModels/Abstract/BookingButtonPageViewModelBase.cs:122
[MonoDroid]   at System.Collections.Concurrent.ObservableConcurrentDictionary`2+<>c__DisplayClass9_0[TKey,TValue].<NotifyObserversOfChange>b__0 (System.Object s) [0x000ba] in /Users/vagrant/Projects/KORONA.shared/KORONA.shared/Extensions/ObservableConcurrentDictionary{TKey,TValue}.cs:52
[MonoDroid]   at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <e98afaf88af5496ea979d99b06e991b9>:0
[MonoDroid]   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <e98afaf88af5496ea979d99b06e991b9>:0
[MonoDroid]   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <e98afaf88af5496ea979d99b06e991b9>:0
[MonoDroid]   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.44(intptr,intptr)
```

**Comment by irongut** [comment]
I'm seeing the same issue when navigating to a new page on Android with Xamarin Forms v4.4.0.991640. The page doing the PushAsync() call has a bindable FlexLayout with several labels in the DataTemplate. Prior to upgrading to v4.4 recently my app was still on v3.6 and had no problems with Fast Renderers.

Android Min Version: 19
Android Target Version: 29
Xamarin Android Version: 28.0.0.3
Xamarin Forms Version: 4.4.0.991640
Visual Studio Verison: 16.4.5

```
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer'.
  at Java.Interop.JniPeerMembers.AssertSelf (Java.Interop.IJavaPeerable self) [0x00029] in <064cde681fba4dd08eaf8b1bdbbfb8c8>:0
  at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeNonvirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x00000] in <064cde681fba4dd08eaf8b1bdbbfb8c8>:0
  at Android.Widget.TextView.set_TextFormatted (Java.Lang.ICharSequence value) [0x00029] in <985f5179c1dd4b3b98fb1b8dcc1ee9be>:0
  at Android.Widget.TextView.set_Text (System.String value) [0x0000d] in <985f5179c1dd4b3b98fb1b8dcc1ee9be>:0
  at Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer.UpdateText () [0x000f3] in D:\a\1\s\Xamarin.Forms.Platform.Android\FastRenderers\LabelRenderer.cs:406
  at Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer.OnElementPropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs e) [0x00123] in D:\a\1\s\Xamarin.Forms.Platform.Android\FastRenderers\LabelRenderer.cs:287
  at (wrapper delegate-invoke) <Module>.invoke_void_object_PropertyChangedEventArgs(object,System.ComponentModel.PropertyChangedEventArgs)
  at Xamarin.Forms.BindableObject.OnPropertyChanged (System.String propertyName) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:229
  at Xamarin.Forms.Element.OnPropertyChanged (System.String propertyName) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:353
  at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) [0x00114] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:461
  at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x00173] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:397
  at Xamarin.Forms.BindingExpression.ApplyCore (System.Object sourceObject, Xamarin.Forms.BindableObject target, Xamarin.Forms.BindableProperty property, System.Boolean fromTarget) [0x00220] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:156
  at Xamarin.Forms.BindingExpression.Apply (System.Boolean fromTarget) [0x0003e] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:52
  at Xamarin.Forms.BindingExpression+BindingExpressionPart.<PropertyChanged>b__49_0 () [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:741
  at Xamarin.Forms.BindingExpression+BindingExpressionPart.PropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs args) [0x000be] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:752
  at Xamarin.Forms.BindingExpression+WeakPropertyChangedProxy.OnPropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs e) [0x00012] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:645
  at (wrapper delegate-invoke) <Module>.invoke_void_object_PropertyChangedEventArgs(object,System.ComponentModel.PropertyChangedEventArgs)
  at EliteALD.ViewModels.BaseViewModel.OnPropertyChanged (System.String propertyName) [0x00011] in D:\Dev\Csharp\ALD\src\EliteALD\EliteALD\EliteALD\ViewModels\BaseViewModel.cs:20
  at EliteALD.ViewModels.PowerDetailViewModel.set_PowerStanding (EliteALD.Models.PowerStanding value) [0x00019] in D:\Dev\Csharp\ALD\src\EliteALD\EliteALD\EliteALD\ViewModels\PowerDetailViewModel.cs:54
  at EliteALD.ViewModels.PowerDetailViewModel.GetPowerDetails (EliteALD.Models.PowerStanding standing) [0x0002e] in D:\Dev\Csharp\ALD\src\EliteALD\EliteALD\EliteALD\ViewModels\PowerDetailViewModel.cs:162
  at EliteALD.ViewModels.StandingsViewModel.OpenPowerDetailsAsync (EliteALD.Models.PowerStanding power) [0x00045] in D:\Dev\Csharp\ALD\src\EliteALD\EliteALD\EliteALD\ViewModels\StandingsViewModel.cs:169
  at EliteALD.ViewModels.StandingsViewModel.<.ctor>b__46_0 (EliteALD.Models.PowerStanding x) [0x0002e] in D:\Dev\Csharp\ALD\src\EliteALD\EliteALD\EliteALD\ViewModels\StandingsViewModel.cs:161
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in /Users/builder/jenkins/workspace/archive-mono/2019-08/android/release/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1021
  at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <985f5179c1dd4b3b98fb1b8dcc1ee9be>:0
  at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <985f5179c1dd4b3b98fb1b8dcc1ee9be>:0
  at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <985f5179c1dd4b3b98fb1b8dcc1ee9be>:0
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.44(intptr,intptr)
```

**Comment by DevEddy** [comment]
For me it happens only on Android with Xamarin.Forms 4.4.0.991640 (reported via AppCenter).

```csharp
Xamarin Exception Stack:
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer'.
  at Java.Interop.JniPeerMembers.AssertSelf (Java.Interop.IJavaPeerable self) <0x7af9b539a0 + 0x000bc> in <0fbf3b1bf489466691c60567f483d8bd>:0
  at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeNonvirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) <0x7af9b54db4 + 0x00033> in <0fbf3b1bf489466691c60567f483d8bd>:0
  at Android.Widget.TextView.set_TextFormatted (Java.Lang.ICharSequence value) <0x7af9b966b0 + 0x000b3> in <149a1f7e8e724849b2c7bc938879af84>:0
  at Android.Widget.TextView.set_Text (System.String value) <0x7af9b96820 + 0x0004b> in <149a1f7e8e724849b2c7bc938879af84>:0
  at Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer.UpdateText () [0x000ff] in <ad1547340995436c904507feabf68321>:0
  at Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer.OnElementPropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs e) [0x00123] in <ad1547340995436c904507feabf68321>:0
  at (wrapper delegate-invoke) <Module>.invoke_void_object_PropertyChangedEventArgs(object,System.ComponentModel.PropertyChangedEventArgs)
  at Xamarin.Forms.BindableObject.OnPropertyChanged (System.String propertyName) <0x7af946803c + 0x0006f> in <64598914a1374c2fa65e161f1f4b5185>:0
  at Xamarin.Forms.Element.OnPropertyChanged (System.String propertyName) <0x7af9478084 + 0x00057> in <64598914a1374c2fa65e161f1f4b5185>:0
  at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) <0x7af9468994 + 0x00207> in <64598914a1374c2fa65e161f1f4b5185>:0
  at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) <0x7af94685c4 + 0x00233> in <64598914a1374c2fa65e161f1f4b5185>:0
  at Xamarin.Forms.BindingExpression.ApplyCore (System.Object sourceObject, Xamarin.Forms.BindableObject target, Xamarin.Forms.BindableProperty property, System.Boolean fromTarget) <0x7af946d1e8 + 0x0055f> in <64598914a1374c2fa65e161f1f4b5185>:0
  at Xamarin.Forms.BindingExpression.Apply (System.Boolean fromTarget) <0x7af946cf14 + 0x00077> in <64598914a1374c2fa65e161f1f4b5185>:0
  at Xamarin.Forms.BindingExpression+BindingExpressionPart.<PropertyChanged>b__49_0 () <0x7af946e8f4 + 0x00013> in <64598914a1374c2fa65e161f1f4b5185>:0
  at Java.Lang.Thread+RunnableImplementor.Run () <0x7af9bb17ac + 0x00047> in <149a1f7e8e724849b2c7bc938879af84>:0
  at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <149a1f7e8e724849b2c7bc938879af84>:0
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.28(intptr,intptr)
```

**Comment by Jerepain** [comment]
Hi,

Same issue in 4.6.0.529-pre3.

I have a simple Grid with 2 Labels in a custom ViewCell in a TableView. When I navigate back and forward 2 times it throws :
```
System.ObjectDisposedException
  Message=Cannot access a disposed object.
Object name: 'Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer'.
```


Same with a Button in a custom ViewCell :
```
System.ObjectDisposedException: 'Cannot access a disposed object.
Object name: 'Xamarin.Forms.Platform.Android.FastRenderers.ButtonRenderer'.'
```
